File: crates/auto-lang/test/a2r/033_multi_delegation/multi_delegation.expected.rs
trait Engine {
    fn start(&self);
}


trait Weapon {
    fn fire(&self);
}


struct WarpDrive {}

impl WarpDrive {
    fn start(&self) {
        println!("WarpDrive engaging");
    }
}

struct LaserCannon {}

impl LaserCannon {
    fn fire(&self) {
        println!("Pew! Pew!");
    }
}

struct Starship {
    core: WarpDrive,
    weapon: LaserCannon,
}

impl Engine for Starship {
    fn start(&self) {
        self.core.start()
    }
}

impl Weapon for Starship {
    fn fire(&self) {
        self.weapon.fire()
    }
}

fn main() {
    let ship: Starship = Starship {};
    ship.start();
    ship.fire();
}

File: crates/auto-lang/test/a2c/13_delegation/002_multi_delegation/multi_delegation.expected.c
#include "multi_delegation.h"



void WarpDrive_Start(struct WarpDrive *self) {
    printf("%s\n", "WarpDrive engaging");
}

void LaserCannon_Fire(struct LaserCannon *self) {
    printf("%s\n", "Pew! Pew!");
}

void Starship_start(struct Starship *self) {
    WarpDrive_start(&self->core);
}
void Starship_fire(struct Starship *self) {
    LaserCannon_fire(&self->weapon);
}

int main(void) {
    struct Starship ship = {};
    Starship_start(&ship);
    Starship_fire(&ship);
    return 0;
}
